### ISSUE TITLE
performance differences of ip between v2.7.3 and v3.1

### ISSUE THREAD

**Issue by zhenwei-intel** [oneapi-src/oneDNN#1640]
Are the following two commands equivalent?

```bash
# benchdnn v2.7.3
numactl -m 0 -C 0-27 ./tests/benchdnn/benchdnn --ip --mode=P --cfg=u8s8u8 --dir=FWD_I --attr-oscale=common:2 mb128ic1024oc1024

# benchdnn v3.1
numactl -m 0 -C 0-27 ./tests/benchdnn/benchdnn --ip --mode=P --cfg=u8s8u8 --dir=FWD_I --attr-scales=src:common:2*+wei:per_oc:1.5*+dst:common:2* mb128ic1024oc1024
```

 If they are equivalent, is there a problem with the performance difference below
- v2.7.3: avg(ms):0.0376083
- v3.1:    avg(ms):0.0391065

(I tested perf of innerproduct between v2.7.3 and v3.1 on clx 8280)

**Comment by dzarukin** [comment]
Hi @zhenwei-intel, there's no pure equivalence between output scale from v2.7 and per argument scales in v3.0.
In your case, dst scale should be stripped off and that will be same from the final result perspective.
v3.0 version precomputes src_scale * wei_scales to mimic output scale behavior for accumulated part.
This introduced an overhead inside the library but allows user drop quantizing bias.
The shape in question is also very small to judge performance impact of scaling.

**Comment by zhenwei-intel** [comment]
Hi @dzarukin, There is a slight difference between the following two examples

```bash
# benchdnn v2.7.3
numactl -C 0-27 ./tests/benchdnn/benchdnn --ip --dir=FWD_I --mode=P --cfg=u8s8f32 --attr-oscale=common:2 mb197ic3072oc768
total perf: min(ms):0.0927734 avg(ms):0.0965185

# benchdnn v3.1
numactl -C 0-27 ./tests/benchdnn/benchdnn --ip --mode=P --cfg=u8s8f32 --dir=FWD_I --attr-scales=src:common:2*+wei:per_oc:1.5* mb197ic3072oc768
total perf: min(ms):0.138184 avg(ms):0.145993

```

**Comment by dzarukin** [comment]
Hi @zhenwei-intel, I think @xuxinzen confirmed this is a regression in the kernel and not due to scales around it, and opened an internal tracker to investigate this deeper. Thank you for reporting it.

**Comment by dzarukin** [comment]
Fixed by [c54f918a](https://github.com/intel-innersource/libraries.performance.math.onednn/commit/c54f918a5a9458e7605cc222160b1b9c5574d634)
